File: docs/snippets/rust/cache/disk_cache.rs
```rust title="disk_cache.rs"
use kreuzberg::{extract_file_sync, ExtractionConfig};

fn main() -> kreuzberg::Result<()> {
    let path = std::env::args()
        .skip(1)
        .find(|a| !a.is_empty() && a != "--")
        .unwrap_or_else(|| "document.pdf".to_string());

    // Enable caching (default: true). The Rust crate uses an internal disk cache.
    let config = ExtractionConfig {
        use_cache: true,
        ..Default::default()
    };

    println!("First extraction (will be cached)...");
    let result1 = extract_file_sync(&path, None, &config)?;
    println!("  - Content length: {}", result1.content.len());

    println!("\nSecond extraction (from cache when available)...");
    let result2 = extract_file_sync(&path, None, &config)?;
    println!("  - Content length: {}", result2.content.len());

    println!("\nResults are identical: {}", result1.content == result2.content);

    Ok(())
}
```

File: docs/snippets/go/cache/disk_cache.go
```go title="disk_cache.go"
package main

import (
	"fmt"
	"log"

	"github.com/xberg-io/xberg/packages/go"
)

func main() {
	useCache := true
	namespace := "documents"
	ttl := uint64(7 * 86400)

	config := xberg.ExtractionConfig{
		UseCache:       &useCache,
		CacheNamespace: &namespace,
		CacheTTLSecs:   &ttl,
	}

	fmt.Println("First extraction (will be cached)...")
	result1, err := xberg.ExtractSync("document.pdf", nil, config)
	if err != nil {
		log.Fatalf("extract failed: %v", err)
	}
	fmt.Printf("  - Content length: %d\n", len(result1.Content))

	fmt.Println("\nSecond extraction (from cache)...")
	result2, err := xberg.ExtractSync("document.pdf", nil, config)
	if err != nil {
		log.Fatalf("extract failed: %v", err)
	}
	fmt.Printf("  - Content length: %d\n", len(result2.Content))

	fmt.Printf("\nResults are identical: %v\n", result1.Content == result2.Content)
}
```
